src/processor/Package.ts

```diff
@@ -85,6 +85,7 @@ export class Package {
     this.removeDateRules();
     this.combineContainsRules();
     this.simplifyOnlyRules(processor);
+    this.makeMappingNamesUnique();
```

Comment:
Uh oh. I haven't looked at your PR to have seen it would be major 😞 . I had opted to do this in the optimizers rather than as each mapping was added because it let me not pass in the existing mappings to the `MappingExtractor`. But I can probably do it there and it would alleviate your conflicts. Do you think it makes more sense in one place or the other?

Reply:
Too late.  Merged. ;-)